plugins/docker/docker_

```diff
@@ -45,44 +46,179 @@ https://docs.docker.com/compose/reference/envvars/):
     env.DOCKER_HOST unix://var/run/docker.sock
 
 =back
+
+=head1 AUTHORS
+
+This section has been reverse-engineered from git logs
+
+* Codimp <contact@lithio.fr>: original rewrite
+* Rowan Wookey <admin@rwky.net>: performance improvement
+* Olivier Mehani <shtrom@ssji.net>: Network support, ClientWrapper, gerenal
+  cleanup
+
+=head1 MAGIC MARKERS
+
+#%# family=auto
+#%# capabilities=autoconf suggest
+
 """
 
 import os
 import sys
-import docker
+from functools import cached_property
 from multiprocessing import Process, Queue
 
+# Early autoconf-style detection
+try:
+    import docker
+except ImportError as e:
+    print(f'no ({e})')
+    sys.exit(1)
```

Comment:
The autoconf specification says, that even in case of `no`, there should be a zero-exitcode.

But the non-zero-exitcode should be emitted in all other (non-`autoconf`) cases, of course.

Thus you should probably just set a `docker_import_success` variable here and use it for a proper `autoconf` later.